julearn/pipeline/tests/test_pipeline_creator.py

```diff
@@ -155,15 +167,228 @@ def test_hyperparameter_tuning(
         X_types=X_types_iris, search_params=search_params
     )
 
+    return pipeline, param_grid
+
+
+def test_hyperparameter_tuning(
+    X_types_iris: Dict[str, List[str]],  # noqa: N803
+    model: str,
+    preprocess: Union[str, List[str]],
+    problem_type: str,
+    get_tuning_params: Callable,
+    search_params: Dict[str, List],
+) -> None:
+    """Test that the pipeline hyperparameter tuning works as expected.
+
+    Parameters
+    ----------
+    X_types_iris : dict
+        The iris dataset features types.
+    model : str
+        The model to test.
+    preprocess : str or list of str
+        The preprocessing steps to test.
+    problem_type : str
+        The problem type to test.
+    get_tuning_params : Callable
+        A function that returns the tuning hyperparameters for a given step.
+    search_params : dict of str and list
+        The parameters for the search.
+
+    Raises
+    ------
+    ValueError
+        If the kind is not supported.
+
+    """
+
+    pipeline, param_grid = _hyperparam_tuning_base_test(
+        X_types_iris,
+        model,
+        preprocess,
+        problem_type,
+        get_tuning_params,
+        search_params,
+    )
     kind = "grid"
     if search_params is not None:
         kind = search_params.get("kind", "grid")
+
     if kind == "grid":
         assert isinstance(pipeline, GridSearchCV)
-        assert pipeline.param_grid == param_grid
+        assert pipeline.param_grid == param_grid  # type: ignore
+    elif kind == "random":
+        assert isinstance(pipeline, RandomizedSearchCV)
+        assert pipeline.param_distributions == param_grid  # type: ignore
     else:
+        raise ValueError(f"Unknown kind {kind}")
```

Comment:
Any specific reason for raising an exception in a test function? Is it requiring something that `pytest.raises()` can provide?